src/classes/USER_UserService.cls

```diff
@@ -0,0 +1,271 @@
+			for(User currentUser : updatedUsers) {
+				userIdToIsFrozen.put(currentUser.Id, userLoginUserIdToIsFrozen.get(currentUser.Id));
+			}
+		}
+
+		return userIdToIsFrozen;
+	}
+	
+
+	/*********************************************************************************************************
+    * @description Handles User changes to IsActive/IsFrozen status by updating the Error Notification Recipients, logging an Error__c, and notifying the org's System Administrators.
+    * @param sysAdminUserIds The User Ids that were modified.
+    * @return void
+    **********************************************************************************************************/
+	@future
+	public static void applySysAdminIsActiveIsFrozenStatusChanges(Set<Id> sysAdminUserIds) {
+		// Sets Error Notification recipient to All Sys Admin
+		setErrorNotificationToAllSysAdmins(sysAdminUserIds);
+
+		// Logs an Error and Emails the Sys Admins
+		handleScheduledJobsSubmittedByInActiveIsFrozenSysAdmin(sysAdminUserIds);
+	}
+
+
+	/*********************************************************************************************************
+    * @description Sets the NPSP Settings Error Notification Recipients to All Sys Admins.
+    * @param sysAdminUserIds The User Ids that were modified.
+    * @return void
+    **********************************************************************************************************/
+	public static void setErrorNotificationToAllSysAdmins(Set<Id> sysAdminUserIds) {
+		Error_Settings__c errorSettings = UTIL_CustomSettingsFacade.getErrorSettings();
+		
+		if(errorSettings.Error_Notifications_To__c.startsWith(OBJECT_ID_PREFIX_USER)) {
+			// Check if one of the sysAdminUserIds is the Error Notification recipient
+			for(Id currentSysAdminId: sysAdminUserIds) {
+				if(errorSettings.Error_Notifications_To__c == currentSysAdminId) {
+					// Set Error Notification recipient to All Sys Admin
+					errorSettings.Error_Notifications_To__c = ERROR_NOTIFICATION_RECIPIENT_ALL_SYS_ADMINS;
+					Database.update(errorSettings, true);
+					break;
+				}
+			}
+		}
+	}
+
+
+	/*********************************************************************************************************
+    * @description Determines if a User is the owner of the Scheduled Jobs.
+    * @param sysAdminUserIds The User Ids that were modified.
+    * @return void
+    **********************************************************************************************************/
+	public static void handleScheduledJobsSubmittedByInActiveIsFrozenSysAdmin(Set<Id> sysAdminUserIds) {
+		Boolean breakOuterCircle = false;
+
+		List<CronTrigger> scheduledJobs = retrieveScheduledJobs(sysAdminUserIds);
+
+		for(Id currentSysAdminId : sysAdminUserIds) {
+			for(CronTrigger eachScheduledJob : scheduledJobs) {
+				if(currentSysAdminId == eachScheduledJob.OwnerId) {
+					notifySysAdminsAboutInActiveIsFrozenSysAdminUser();
+					breakOuterCircle = true;
+					break;
```

Comment:
using the boolean to break out of two for loops is a bit non-standard.  if you are really trying to stop processing that method, you should just use a return.  I do wonder, however, whether you should be ensuring to verify all  of the passed in admins, not just stopping at the first?

Reply:
I agree about the usage of the boolean. I hesitated about using return since I was in a void method. However, I like using return in this case and have updated the code.